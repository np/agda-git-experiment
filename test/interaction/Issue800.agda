-- Andreas, 2013-02-18 problem with 'with'-display, see also issue 295
-- {-# OPTIONS -v tc.with:50 #-}
module Issue800 where

  data ⊤ : Set where
    tt : ⊤

  data I⊤ : ⊤ → Set where
    itt : ∀ r → I⊤ r

  bug : ∀ l → ∀ k → I⊤ l → ⊤
  bug .l k (itt l) with itt k
  ... | foo = {! foo!}

{-
Current rewriting:
  bug .l l (itt k) | itt .k = ?

Desired rewriting:
  bug .l k (itt l) | itt .k = ?
-}
