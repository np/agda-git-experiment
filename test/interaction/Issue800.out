Agda2> (agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking Issue800 (Issue800.agda).\n" t)
(agda2-info-action "*Type-checking*" "Finished Issue800.\n" t)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : âŠ¤\n" nil)
((last . 1) . (agda2-goals-action '(0)))
Agda2> ((last . 2) . (agda2-make-case-action '("bug .l k (itt l) | itt .k = ?")))
((last . 1) . (agda2-goals-action '(0)))
Agda2> 